internal/util.go

```diff
@@ -12,13 +12,17 @@ import (
 // for resources that can contain an arbitrary or dynamic number of fields.
 func RemainingKeys(s interface{}, m map[string]interface{}) (extras map[string]interface{}) {
 	extras = make(map[string]interface{})
+	for k, v := range m {
+		extras[k] = v
+	}
+
 	valueOf := reflect.ValueOf(s)
 	typeOf := reflect.TypeOf(s)
 	for i := 0; i < valueOf.NumField(); i++ {
 		field := typeOf.Field(i)
 		tagValue := field.Tag.Get("json")
-		if _, ok := m[tagValue]; !ok {
-			extras[tagValue] = m[tagValue]
+		if _, ok := extras[tagValue]; ok {
+			delete(extras, tagValue)
```

Comment:
Example:

You have this response
```json
{
  "user_id": "abcd1234",
  "username": "John",
  "status": "current",
  "custom_field": "foo",
  "is_admin": true,
}
```
and this object to unmarshal it into:
```go
type User struct {
  UserID string `json:"user_id"`
  Username string `json:"username"`
  Status string
  IsAdmin bool
}
```

Both `custom_field` and `is_admin` should be in `Extra`.

Logic would be something like

```
for each struct field {
  if field has a struct tag {
    if struct tag is "-" {
      skip field
    } else {
      check map keys for struct tag
    }
  } else {
    check map keys for lower-cased field name
  }
}
```